File: hrms_employee/emptype/empTypeService/empTypeService.py
from hrms_employee.models import employee_type


class empType:
    @classmethod
    def saveEmpType(cls,empTypeName,created_date,created_by):
        try:
            saveqry = employee_type(type_name=empTypeName,created_date=created_date,created_by=created_by)
            saveqry.save()
            successobj = {'data':'success'}
            return successobj

        except Exception, err:
            failureobj = {
                'response': "Failure"
            }
            return failureobj

    @classmethod
    def listingEmpType(cls):
        try:
            listqry = employee_type.objects.all()
            datalist = []
            for values in listqry:
                datalist.append({
                    'emp_type_id' : values.emp_typeId,
                    'type_name' : values.type_name,
                })
            return datalist

        except Exception, err:
            failureobj = {
                'response': "Failure"
            }
            return failureobj

    @classmethod
    def deleteEmpType(cls,typeid):
        try:
            delqry = employee_type.objects.get(emp_typeId=typeid)
            delqry.delete()
            dataobj = {'data':'success'}
            return dataobj

        except Exception, err:
            failureobj = {
                'response': "Failure"
            }
            return failureobj

    @classmethod
    def getEmpTypedata(cls, typeid):
        try:
            getdataqry = employee_type.objects.get(emp_typeId=typeid)

            dataobj = {
                'id' : getdataqry.emp_typeId,
                'type_name' : getdataqry.type_name,
            }

            return dataobj

        except Exception, err:
            failureobj = {
                'response': "Failure"
            }
            return failureobj

    @classmethod
    def updateEmpTypedata(cl,typeId,typeName,updated_date,updated_by):
        try:
            updateqry = employee_type.objects.get(emp_typeId=typeId)

            updateqry.type_name = typeName
            updateqry.updated_date=updated_date
            updateqry.updated_by=updated_by

            updateqry.save()

            dataobj = {'data':'success'}

            return dataobj

        except Exception, err:
            failureobj = {
                'response': "Failure"
            }
            return failureobj

File: static/components/employee/components/employeetype/empTypeService/empTypeService.js
angular.module('employeeApp.empTypeService',[])
.service('empTypeService',['$http',function($http){

    employeeType = {}

    employeeType.saveEmpType = function(empTypeName, success, failure){

        $http.post('/employee/api/saveemptype/',{
            'emptypename' : empTypeName,
        }).then(success,failure)
    }

    employeeType.listingEmpType = function(success, failure){
        $http.get('/employee/api/listemptype/').then(success,failure)
    }

    employeeType.delEmpType = function(id,success, failure){
        $http.delete('/employee/api/delemptype/?id='+id).then(success,failure)
    }

    employeeType.getdataEmpType = function(id,success, failure){
        $http.get('/employee/api/getemptype/?id='+id).then(success,failure)
    }

    employeeType.updateEmpType = function(id,typeName,success, failure){
        $http.post('/employee/api/updateemptype/',{
            'id' : id,
            'emptypename' : typeName,
        }).then(success,failure)
    }

    return employeeType

}])